### PR TITLE
Enhance Email notification according to feedback

### PR DESCRIPTION
**Pull request by jinchihe**
**Which issue is resolved by this Pull Request:** 
Resolves #
Enhance Email notification according to feedbacks:

1. Fix traceback  problem if no attachment.
2. Support multiple attachments.
3. Better to print some logs when sending Email.
4. Inputs `server` may be confuse user to set Email server, but in fact that server secret, update that.
5. Email canot be got if the `--recipients` comma separated without space.
6. Inputs some wrong server address or that cannnot be connected, better to print reasonable message. Do not like Traceback but error or warning message.

**Description of your changes:**

**Environment tested:**

* Python Version (use `python --version`):
* Tekton Version (use `tkn version`):
* Kubernetes Version (use `kubectl version`):
* OS (e.g. from `/etc/os-release`):

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

- [ ] Do you want this pull request (PR) cherry-picked into the current release branch?

    [Learn more about cherry-picking updates into the release branch](https://github.com/kubeflow/pipelines/blob/master/RELEASE.md#cherry-picking-pull-requests-to-release-branch).

### CODE REVIEW

File: components/notification/component.yaml [hunk 1]
```diff
@@ -36,40 +42,44 @@ implementation:
...
+          encoders.encode_base64(part)
+          part.add_header('Content-Disposition',
+                          'attachment; filename="{}"'.format(Path(path).name))
+          msg.attach(part)
       for receiver in receiver_emails.split(' '):
+          logging.info("Sending Email to {}".format(receiver))
           server.sendmail(sender_email, receiver, msg.as_string())
       server.quit()
     args: [
-      --server, {inputValue: server},
```
Comment (Tomcli) [inline components/notification/component.yaml:69]: do you need to map the `server-secret` over here?

File: components/notification/component.yaml [hunk 2]
```diff
@@ -36,40 +42,44 @@ implementation:
...
       --subject, {inputValue: subject},
       --body, {inputValue: body},
       --sender, {inputValue: sender},
       --recipients, {inputValue: recipients},
       --attachment_path, {inputValue: attachment_path},
```
Comment (Tomcli) [inline components/notification/component.yaml:81]: to reflect your changes

```suggestion
      --attachment-path, {inputValue: attachment_path},
```

File: components/notification/component.yaml (hunk 1 above)
Comment (jinchihe) [reply components/notification/component.yaml:69]: Yeah. Fixed, thanks.

**Comment by Tomcli** [comment]
thanks @jinchihe 
/lgtm
/approve
/hold
Since your PR is still titled WIP, you can unhold this PR once you are ready with all your changes.

**Comment by jinchihe** [comment]
@Tomcli Thanks a lot! I marked to wip to wait more feedback, seems that's ready to go now :-)

**Pull request merged**